packages/playground/src/portal/twin_view.vue

```diff
@@ -1,71 +1,121 @@
-      </v-list>
-      <v-card-actions v-if="updateRelay" class="justify-end mx-4 mb-4">
-        <v-btn class="custom-button bg-primary" @click="editTwin">Edit</v-btn>
-      </v-card-actions>
-    </v-card>
-  </v-container>
+
+    <v-container class="custom-container">
+      <v-card color="primary" class="d-flex justify-center items-center mt-3 pa-3 text-center">
+        <v-icon size="30" class="pr-3">mdi-account-supervisor-outline</v-icon>
+        <v-card-title class="pa-0">Twin Details</v-card-title>
+      </v-card>
+
+      <v-card>
+        <v-list class="custom-list">
+          <v-row>
+            <v-col cols="1" sm="2" class="column-style my-4">
+              <v-list-item> ID </v-list-item>
+              <v-divider></v-divider>
+              <v-list-item> Address </v-list-item>
+              <v-divider></v-divider>
+              <v-list-item> Relay </v-list-item>
+            </v-col>
+            <v-col cols="1" sm="10" class="my-4">
+              <v-list-item> {{ profileManager.profile?.twinId.toString() }} </v-list-item>
+              <v-divider></v-divider>
+              <v-list-item>
+                <div style="display: flex; justify-content: space-between; align-items: center">
+                  <span>{{ profileManager.profile?.address }}</span>
+                  <v-icon @click="copy(profileManager.profile?.address as string)"> mdi-content-copy </v-icon>
+                </div>
+              </v-list-item>
+              <v-divider></v-divider>
+              <v-list-item> {{ profileManager.profile?.relay }} </v-list-item>
+            </v-col>
+          </v-row>
+        </v-list>
+        <v-card-actions v-if="updateRelay" class="justify-end mx-4 mb-4">
+          <v-btn class="custom-button bg-primary" @click="editTwin">Edit</v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-container>
+  </div>
 </template>
 
 <script lang="ts" setup>
 import { generatePublicKey } from "@threefold/rmb_direct_client";
+import type { Proposal } from "@threefold/tfchain_client";
 import { onMounted, ref } from "vue";
 
+import router from "../router";
 import { useProfileManager } from "../stores";
+import type { Farm } from "../types";
 import { createCustomToast, ToastType } from "../utils/custom_toast";
+import { getFarms } from "../utils/get_farms";
 import { getGrid } from "../utils/grid";
-
 const profileManager = useProfileManager();
 
 const editingTwin = ref(false);
 const relay = ref(profileManager.profile?.relay || "");
 const updateRelay = ref(false);
 const errorMsg = ref("");
+const openVotePopup = ref(false);
+const numberOfProposalsToVoteOn = ref(0);
+const userFarms = ref<Farm[]>();
+onMounted(async () => {
+  const profile = profileManager.profile!;
+  const grid = await getGrid(profile);
+  if (!grid) {
+    return;
+  }
+  userFarms.value = await getFarms(grid, { ownedBy: profile.twinId }, {});
+  if (!userFarms.value.length) {
+    return;
+  }
+  const proposals = grid?.dao.get();
```

Comment:
The question mark is unnecessary, given the condition you've set for initializing the grid.